torch/_decomp/decompositions.py

```diff
@@ -460,6 +460,26 @@ def nll_loss_backward(
     return grad_input * grad_output
 
 
+@register_decomposition(aten.binary_cross_entropy)
+def binary_cross_entropy(
+    self: Tensor,
+    target: Tensor,
+    weight: Optional[Tensor] = None,
+    reduction: int = Reduction.MEAN.value,
+) -> Tensor:
+    # We cannot currently model this without introducing data-dependent control flow
+    # TORCH_CHECK(
+    #     (input_val >= 0) && (input_val <= 1),
+    #     "all elements of input should be between 0 and 1"
+    # )
```

Comment:
Nit: It's not clear which of {self, target, weight} is "input_val", but I guess it doesn't matter much until we need to figure out how to model this